tests/performance_tests/log_reader.py

```diff
@@ -233,6 +251,40 @@ def populateTrxSentTimestamp(trxSent: dict, trxDict: dict, notFound):
         else:
             notFound.append(sentTrxId)
 
+def getProductionWindows(prodDict: dict, blockDict: dict, data: chainData):
+    prod = ""
+    count = 0
+    blocksFromCurProd = 0
+    numProdWindows = 0
+    for k, v in blockDict.items():
+        count += 1
+        if prod == "":
+            prod = v.producer
+        if prod != v.producer or count+data.startBlock == data.ceaseBlock:
+            prodDict[str(numProdWindows)] = productionWindow(prod, count-blocksFromCurProd+data.startBlock-1, count+data.startBlock-2, blocksFromCurProd)
+            prod = v.producer
+            blocksFromCurProd = 1
+            numProdWindows += 1
+        else:
+            blocksFromCurProd += 1
+    return prodDict
+
+def calcProductionWindows(prodDict: dict):
+    prodWindows = productionWindows()
+    prodWindows.totalWindows = len(prodDict) - 2
+    totalBlocksForAverage = 0
+    for k, v in prodDict.items():
+        if k != "0" and k != str(prodWindows.totalWindows+1):
+            if v.blockCount < 12:
```

Comment:
Make `12` a constant. There is a possibility it might change in the future or a sister chain might want a different value.